x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/deployment/DeploymentManager.java

```diff
@@ -142,7 +145,9 @@ private void doStartDeployment(TrainedModelDeploymentTask task, ActionListener<T
         });
 
         ActionListener<Boolean> modelLoadedListener = ActionListener.wrap(success -> {
-            executorServiceForProcess.execute(() -> processContext.getResultProcessor().process(processContext.process.get()));
+            executorServiceForProcess.execute(
+                () -> processContext.getResultProcessor().process(processContext.process.get().readResults())
```

Comment:
Now that `readResults()` is not called in the try-catch block in the result processor, this could throw here and it'll be unhandled.

Reply:
++, even worse is that `readResults()` will block until the first token has been read. 

I made the change to make unit testing easier but alternative of mocking `nativeProcess::readResults()` was very simple.